### ISSUE TITLE
[WC 4.8 beta + WP 5.6 beta4] Fatal Error when testing with Gutenberg + Creative Mail by Constant Contact + Product Bundles: allowed memory size

### ISSUE THREAD

**Issue by juliaamosova** [woocommerce/woocommerce#28323]
**Prerequisites (mark completed items with an [x]):**
- [x] I have have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

**Describe the bug**

I tested WC 4.8 beta with WP 5.6 beta4 (with PHP 7.4.12) on JN site + Gutenberg plugin +  Creative Mail by Constant Contact (offered to be installed during OBW) + Product Bundles and came across the following Fatal Errors:

```
Fatal error: Allowed memory size of 268435456 bytes exhausted (tried to allocate 20480 bytes) in /srv/users/userde216576/apps/userde216576/public/wp-includes/class-wp-hook.php on line 280
```

```
Fatal error: Allowed memory size of 268435456 bytes exhausted (tried to allocate 20480 bytes) in /srv/users/userde216576/apps/userde216576/public/wp-includes/class-wp-fatal-error-handler.php on line 72
```

The site continued to work, however.

![Screen Shot 2020-11-17 at 10 37 35 AM](https://user-images.githubusercontent.com/19143190/99412041-4d43fd80-28c2-11eb-94f2-2dc66a331663.png)

**Expected behavior**

I expected did not encounter Fatal Error on the site at any point.

**Steps to reproduce the bug:**

Steps to reproduce the bug:

1. Create a new JN site with WP 5.6 (through WP Beat Tester plugin) and WC 4.8 beta + Gutenberg plugin.
2. Complete OBW with a US-based store and `Fashion` as an option of the store industry. Select to install free business extensions (which is the step when `Creative Mail by Constant Contact ` plugin gets installed).
3. Also make sure to select additional product types (Product Bundles) during OBW. 
4. Once on Home Screen, complete the task to `Purchase & Install extensions` in order to get Product Bundles installed.
5. Once done, observe errors. 

**Screenshots**

See above. 

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
We use the [WooCommerce System Status Report](https://docs.woocommerce.com/document/understanding-the-woocommerce-system-status-report/) to help us evaluate the issue. 
Without this report we won't be able to fully evaluate this issue.
<details>
```
`
### WordPress Environment ###

WordPress address (URL): https://low-bird.jurassic.ninja
Site address (URL): https://low-bird.jurassic.ninja
WC Version: 4.8.0
REST API Version: ✔ 4.8.0
WC Blocks Version: ✔ 3.8.0
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 1.7.0
Log Directory Writable: ✔
WP Version: 5.6-beta4-49619
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.46 (Unix) OpenSSL/1.0.2g
PHP Version: 7.4.12
PHP Post Max Size: 1 GB
PHP Time Limit: 30
PHP Max Input Vars: 5000
cURL Version: 7.47.0
OpenSSL/1.0.2g

SUHOSIN Installed: –
MySQL Version: 5.7.32-0ubuntu0.16.04.1-log
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 4.8.0
WC Database Prefix: wp_
Total Database Size: 5.56MB
Database Data Size: 4.01MB
Database Index Size: 1.55MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailchimp_carts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailchimp_jobs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_options: Data: 3.03MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_bookings_availability: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_bookings_availabilitymeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_booking_relationships: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_bundled_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_bundled_items: Data: 0.02MB + Index: 0.03MB + Engine InnoDB

### Post Type Counts ###

attachment: 1
page: 7
post: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ❌Error messages should not be shown to visitors.

### Active Plugins (7) ###

Companion Plugin: by Osk – 1.17
Creative Mail by Constant Contact: by Constant Contact – 1.2.1 – Installed version not tested with active version of WooCommerce 4.0
Gutenberg: by Gutenberg Team – 9.3.0
Jetpack by WordPress.com: by Automattic – 9.1
WooCommerce Product Bundles: by SomewhereWarm – 6.5.1
WooCommerce: by Automattic – 4.8.0-beta.1
WordPress Beta Tester: by Peter Westwood
Andy Fragen – 3.0.3


### Inactive Plugins (11) ###

Akismet Anti-Spam: by Automattic – 4.1.7
Facebook for WooCommerce: by Facebook – 2.1.4
Google Ads & Marketing by Kliken: by Kliken – 1.0.6
Hello Dolly: by Matt Mullenweg – 1.7.2
Mailchimp for WooCommerce: by Mailchimp – 2.4.7
WooCommerce Bookings: by WooCommerce – 1.15.30
WooCommerce Memberships: by SkyVerge – 1.19.2
WooCommerce Payments: by Automattic – 1.6.0
WooCommerce Product Add-ons: by WooCommerce – 3.1.1
WooCommerce Shipping & Tax: by Automattic – 1.25.2
WooCommerce Subscriptions: by WooCommerce – 3.0.10

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: bundle (bundle)
external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔

### WC Pages ###

Shop base: #7 - /shop/
Cart: #8 - /cart/
Checkout: #9 - /checkout/
My account: #10 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 2.9.0
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Product Bundles ###

Database Version: 6.5.1
Loopback Test: ✔
Template Overrides: –

### Action Scheduler ###

Complete: 11
Oldest: 2020-11-17 14:55:38 +0000
Newest: 2020-11-17 15:52:35 +0000

`
```
</details>

**Comment by timmyc** [comment]
@juliaamosova do you happen to know which version of Gutenberg was installed? I saw p9o2xV-19h-p2 yesterday so wasn't sure if it might be related or not.

**Comment by juliaamosova** [comment]
@timmyc yep, I have Gutenberg 9.3.0 installed on the site where the error happened.

**Comment by juliaamosova** [comment]
Noting here that if I deactivate WooCommerce but keep Gutenberg and Creative Mail by Constant Contact plugins active, I am still getting the same errors:

```
Fatal error: Allowed memory size of 268435456 bytes exhausted (tried to allocate 262144 bytes) in /srv/users/userde216576/apps/userde216576/public/wp-includes/class-wp-hook.php on line 287
```

```
Fatal error: Allowed memory size of 268435456 bytes exhausted (tried to allocate 262144 bytes) in Unknown on line 0
```

**Comment by timmyc** [comment]
I did a similar test on atomic ephmeral and haven't been able to reproduce this myself yet.

**Comment by tammullen** [comment]
I can't reproduce. I suspect is due to the same Gutenberg 9.3.0 issue in https://github.com/woocommerce/woocommerce/issues/28321 which is fixed in 9.4.0

**Comment by tammullen** [comment]
Closing after further discussion we believe this to be related to Gutenberg 9.3.0
